app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -521,6 +564,60 @@ def run_operation(parameters, operation_name = nil)
       end
     end
 
+    NotificationArgs = Struct.new(:type, :operation_name, :operation_args,
+                                  :target_resource, :entity_klass, :detailed_message) do
+      def event_type(mw_entity)
+        '%{entity_type}.%{operation}.%{status}' %
+          { :entity_type => mw_entity.kind_of?(MiddlewareServer) ? 'MwServer' : 'MwDomain',
```

Comment:
I think this is `entity_klass`, no?

Reply:
You made me doubt for a moment. But it's ok to check `mw_entity`. `entity_klass` can be, for example, a `MiddlewareDatasource`, and `mw_entity` will be the owning `MiddlewareServer` server.